appender.go

```diff
@@ -72,7 +72,7 @@ type colInfo struct {
 
 // Appender holds the DuckDB appender. It allows efficient bulk loading into a DuckDB database.
 type Appender struct {
-	c        *conn
+	c        *connection
```

Comment:
I'm not married to `conn` but I've thought we could use the standard abbreviation in Go for a connection.

https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/net/iprawsock.go;l=99
https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/net/tcpsock.go;l=112
https://cs.opensource.google/go/x/crypto/+/refs/tags/v0.21.0:ssh/client.go;l=20

Reply:
Fair point. We also have it in `driver.Conn`. I was mainly trying to unify this, as I was getting confused by our different naming conventions: `connection, con, conn, c, Connector...`. 

Maybe we can unify it like this?
```Go
type conn
var con conn

type Connector
var c Connector
```

Reply:
I've reverted the changes, but I'd love it if we could slowly move towards more uniform names for this situation in future PRs. :)